.github/workflows/maestro_deploy-tests.yaml

```diff
@@ -0,0 +1,47 @@
+name: Maestro Tests
+on:
+  push:
+    branches: [ 'main' ]
+    paths:
+    - 'maestro/**'
+  pull_request:
+    branches: [ 'main' ]
+    paths:
+    - 'maestro/**'
+jobs:
+  run-deploy-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Kind
+        run: |
+          curl -Lo ./kind "https://github.com/kubernetes-sigs/kind/releases/download/v0.20.0/kind-$(uname)-amd64"
+          chmod +x ./kind
+          sudo mv kind /usr/local/bin
+          which kind
+      - name: Create Kind cluster
+        run: |
+          kind create cluster --config maestro/tests/integration/deploys/kind-config.yaml 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Setup poetry
+        run: |
+          pipx ensurepath
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
+          pipx install poetry
+          poetry self add poetry-plugin-shell
+      - name: Install dependencies
+        run: |
+          cd maestro
+          poetry install
+      - name: Run unit tests
+        run: |
+          cd maestro
+          export IMAGE_PUSH_CMD='kind load docker-image docker.io/library/maestro:latest'
+          export IMAGE_TAG_CMD='docker tag localhost/maestro:latest docker.io/library/maestro:latest'
+          export DEPLOY_DOCKER_TEST=1
+          export DEPLOY_KUBERNETES_TEST=1
+          export IN_GITHUB_ACTION=1
+          poetry run pytest
```

Comment:
Please specify just the deploy folder since otherwise this will run all the tests which is not needed as we have a different GH action workflow for that